src/platform/datapath_raw_xdp_win.c

```diff
@@ -107,246 +108,137 @@ CxPlatXdpExecute(
-            hRes,
-            "ConnectServer");
-        goto Cleanup;
-    }
+        Status = Xdp->XdpApi->XskSetSockopt(TxXsk, XSK_SOCKOPT_UMEM_REG, &TxUmem, sizeof(TxUmem));
+        if (QUIC_FAILED(Status)) { CloseHandle(TxXsk); return Status; }
 
-    // Step 4: --------------------------------------------------
-    // Set security levels on the proxy -------------------------
-    hRes = CoSetProxyBlanket(
-       (IUnknown*)pSvc,             // Indicates the proxy to set
-       RPC_C_AUTHN_WINNT,           // RPC_C_AUTHN_xxx
-       RPC_C_AUTHZ_NONE,            // RPC_C_AUTHZ_xxx
-       NULL,                        // Server principal name
-       RPC_C_AUTHN_LEVEL_CALL,      // RPC_C_AUTHN_LEVEL_xxx
-       RPC_C_IMP_LEVEL_IMPERSONATE, // RPC_C_IMP_LEVEL_xxx
-       NULL,                        // client identity
-       EOAC_NONE                    // proxy capabilities
-    );
-    if (FAILED(hRes)) {
-        QuicTraceEvent(
-            LibraryErrorStatus,
-            "[ lib] ERROR, %u, %s.",
-            hRes,
-            "CoSetProxyBlanket");
-        goto Cleanup;
-    }
+        Status = Xdp->XdpApi->XskSetSockopt(TxXsk, XSK_SOCKOPT_TX_RING_SIZE, &TxRingSize, sizeof(TxRingSize));
+        if (QUIC_FAILED(Status)) { CloseHandle(TxXsk); return Status; }
 
-    // Step 5: --------------------------------------------------
-    // Use the IWbemServices pointer to make requests of WMI ----
-    wchar_t query[512] = { '\0' };
-    (void)wcscat_s(query, 512, L"SELECT * FROM MSFT_NetAdapterRssSettingData WHERE Name='");
-    (void)wcscat_s(query, 512, if_alias);
-    (void)wcscat_s(query, 512, L"'");
-    //AF_XDP_LOG(INFO, "WMI query = \"%ws\"\n", query);
-
-    BSTR Language = SysAllocString(L"WQL");
-    BSTR Query = SysAllocString(query);
-    hRes = pSvc->lpVtbl->ExecQuery(pSvc,
-        Language,
-        Query,
-        WBEM_FLAG_FORWARD_ONLY,         // Flags
-        0,                              // Context
-        &pEnum
-        );
-    SysFreeString(Query);
-    SysFreeString(Language);
-    if (FAILED(hRes)) {
-        QuicTraceEvent(
-            LibraryErrorStatus,
-            "[ lib] ERROR, %u, %s.",
-            hRes,
-            "ExecQuery");
-        goto Cleanup;
-    }
+        Status = Xdp->XdpApi->XskSetSockopt(TxXsk, XSK_SOCKOPT_TX_COMPLETION_RING_SIZE, &TxRingSize, sizeof(TxRingSize));
+        if (QUIC_FAILED(Status)) { CloseHandle(TxXsk); return Status; }
 
-    // Step 6: -------------------------------------------------
-    // Get the data from the query in step 6 -------------------
-    IWbemClassObject *pclsObj = NULL;
-    ULONG uReturn = 0;
-    while (pEnum) {
-        HRESULT hr = pEnum->lpVtbl->Next(pEnum, WBEM_INFINITE, 1,
-            &pclsObj, &uReturn);
-
-        if (0 == uReturn) {
+        uint32_t Flags = XSK_BIND_FLAG_TX;
+        Status = Xdp->XdpApi->XskBind(TxXsk, InterfaceIndex, i, Flags);
+        if (QUIC_FAILED(Status)) { // No more queues. Break out.
```

Comment:
there could be other errors. probably worth logging explicitly.

Reply:
You want me to just log this one?

Reply:
this and the "activate" are the only two that might fail in nontrivial and unpredictable ways. it's probably worth logging them all if you have time.